test/globalize_test.rb

```diff
@@ -217,4 +217,28 @@ class GlobalizeTest < MiniTest::Spec
       end
     end
   end
+
+  describe 'with locale' do
+    def perform_with_locale(locale)
+      Thread.new do
+        Globalize.with_locale(locale) do
+          sleep 1
```

Comment:
I see what you mean.
With ivar and sleep: it fails, but when I remove the sleep - it suddenly passes.
The best news is that it also fails with `Thread.pass` and ivars, but passes with Thread.current.
I suppose it means that we can replace sleep with Thread.pass :)